enhancements/scheduling/scheduler-profiles.md

```diff
@@ -0,0 +1,208 @@
+---
+title: scheduling-profiles
+authors:
+  - "@damemi"
+reviewers:
+  - "@maszulik"
+  - "@ingvagabund"
+  - "@deads2k"
+approvers:
+  - "@maszulik"
+  - "@ingvagabund"
+  - "@deads2k"
+creation-date: 2020-11-11
+last-updated: 2020-11-11
+status: provisional
+see-also:
+  - "https://github.com/kubernetes/enhancements/blob/master/keps/sig-scheduling/624-scheduling-framework/kep.yaml"
+  - "/enhancements/kube-apiserver/audit-policy.md"
+replaces:
+superseded-by:
+---
+
+# Scheduling Profiles
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement proposes a new field in the `schedulers.config.openshift.io/v1` API to allow
+configuration of enabled and disabled scheduler plugins via certain defined "profiles".
+
+This will allow users to configure pod scheduling beyond the default set of enabled plugins. It
+is effectively the equivalent to current scheduler Policy config, which has been deprecated.
+
+## Motivation
+
+As of Kubernetes 1.19, plugin profile configuration has replaced Policy files for the scheduler.
+We currently support setting scheduler predicates and priorities through the Policy API, and need
+to offer similar configuration with the new API in order to maintain a level of feature parity.
+
+However, the new plugin API is more expansive than the old API and provides more configuration possibilities
+than is necessary for most customers or reasonable to support. For those reasons we propose an
+abstracted representation of the most common plugin configurations for users to configure.
+
+### Goals
+
+1. define several common scheduler plugin configurations
+2. add a `profile` API field to `schedulers.config.openshift.io/v1` to specify which predetermined
+profile to use
+3. enable these profiles by kube-scheduler-operator
+
+### Non-Goals
+
+* configuration of individual scheduler plugins
+* platform support for custom schedulers or out-of-tree plugins
```

Comment:
How does the restricted list of predefined profiles align with support for custom scheduler if there's an absence of knowing in advance if the operator profiles are going to be applicable for the custom scheduler?

Reply:
Why pre-defined profiles rule out the possibility of using a custom scheduler? Even thought the profiles will be hardcoded it'll be pretty straightforward to see what they do. See https://github.com/openshift/cluster-kube-apiserver-operator/blob/master/bindata/v4.1.0/kube-apiserver/audit-policies-cm.yaml which is how we keep the pre-defined profiles for auditing. The exact same mechanics will be used for scheduler profiles.

Reply:
If you build a custom scheduler in hyperkube, you could in theory pass that through the operator with the $IMAGE env var. But you won't be able to enable any custom plugins you compiled into that scheduler, so the only real use case of that would be re-writing default plugins. At that point the user might as well just deploy the custom scheduler as a 2nd scheduler, and it's clearer to say we just don't support custom plugins in the default scheduler.